lms/djangoapps/teams/static/teams/js/views/teams_tab.js

```diff
@@ -10,259 +10,333 @@
+                        this.getTopic(topicID)
+                            .done(function(topic) {
+                                var collection = new TeamCollection([], {
+                                    course_id: self.courseID,
+                                    topic_id: topicID,
+                                    url: self.teamsUrl,
+                                    per_page: 10
+                                });
+                                collection.goTo(1)
+                                    .done(function() {
+                                        var teamsView = new TeamsView({
+                                            router: router,
+                                            topic: topic,
+                                            collection: collection,
+                                            maxTeamSize: self.maxTeamSize
+                                        });
+                                        deferred.resolve(self.createViewWithHeader(teamsView, topic));
+                                    });
+                            });
+                    }
+                    return deferred.promise();
+                },
 
-                   /**
-                    * Return a promise for the TeamsView for the given
-                    * topic ID.
-                    */
-                   getTeamsView: function (topicID) {
-                       // Lazily load the teams-for-topic view in
-                       // order to avoid making an extra AJAX call.
-                       if (!_.isUndefined(this.teamsView)
-                               && this.teamsView.main.collection.topic_id === topicID) {
-                           return this.identityPromise(this.teamsView);
-                       }
-                       var self = this,
-                           teamCollection = new TeamCollection([], {
-                               course_id: this.course_id,
-                               url: this.teams_url,
-                               topic_id: topicID,
-                               per_page: 10
-                           }),
-                           teamPromise = teamCollection.goTo(1).fail(function (xhr) {
-                               if (xhr.status === 400) {
-                                   self.topicNotFound(topicID);
-                               }
-                           }),
-                           topicPromise = this.getTopic(topicID).fail(function (xhr) {
-                               if (xhr.status === 404) {
-                                   self.topicNotFound(topicID);
-                               }
-                           });
-                       return $.when(topicPromise, teamPromise).pipe(
-                           _.bind(this.constructTeamView, this)
-                       );
-                   },
+                /**
+                 * Render the list of teams for the given topic ID.
```

Comment:
Documentation is wrong.

And why is this browseTeam? Aren't we talking about a single team here?